rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/server/ImportNodeSourceServlet.java

```diff
@@ -0,0 +1,130 @@
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * If needed, contact us to obtain a release under GPL Version 2 or 3
+ * or a different license than the AGPL.
+ */
+package org.ow2.proactive_grid_cloud_portal.rm.server;
+
+import static org.ow2.proactive_grid_cloud_portal.rm.server.NSCreationServlet.MAX_UPLOAD_SIZE;
+
+import java.io.*;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.List;
+
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.apache.commons.fileupload.FileItem;
+import org.apache.commons.fileupload.disk.DiskFileItemFactory;
+import org.apache.commons.fileupload.servlet.ServletFileUpload;
+import org.codehaus.jettison.json.JSONException;
+import org.codehaus.jettison.json.JSONObject;
+import org.codehaus.jettison.json.JSONWriter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gwt.core.client.GWT;
+
+
+/**
+ * Servlet invoked when a node source is imported from a file. The request
+ * content is expected to be in JSON format, containing the configuration of a
+ * node source.
+ */
+public class ImportNodeSourceServlet extends HttpServlet {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ImportNodeSourceServlet.class);
+
+    private static final String NODE_SOURCE_FILE_NAME_SUFFIX = "-configuration.json";
+
+    public static final String SERVLET_MAPPING = "importnodesource";
+
+    public static final String MAIN_FORM_ITEM_NAME = "nodeSourceJson";
+
+    public static final String NODE_SOURCE_NAME_KEY = "nodeSourceName";
+
+    @Override
+    protected void doPost(HttpServletRequest request, HttpServletResponse response) {
+        readNodeSourceConfigurationFile(request, response);
+    }
+
+    @Override
+    protected void doGet(HttpServletRequest request, HttpServletResponse response) {
+        readNodeSourceConfigurationFile(request, response);
+    }
+
+    private void readNodeSourceConfigurationFile(HttpServletRequest request, HttpServletResponse response) {
+        response.setContentType("text/html");
+
+        try {
+            DiskFileItemFactory factory = new DiskFileItemFactory();
+            factory.setSizeThreshold(4096);
+            factory.setRepository(new File(System.getProperty("java.io.tmpdir")));
+            ServletFileUpload upload = new ServletFileUpload(factory);
+            upload.setSizeMax(MAX_UPLOAD_SIZE);
+
+            List<FileItem> fileItems = upload.parseRequest(request);
+            Iterator<?> i = fileItems.iterator();
+            while (i.hasNext()) {
+
+                FileItem item = (FileItem) i.next();
+                String name = item.getFieldName();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this useless assignment to local variable "name". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1854)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this unused "name" local variable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1481)